src/components/Section/Identification/ApplicantName/ApplicantName.jsx

```diff
@@ -2,18 +2,66 @@ import React from 'react'
 import { i18n } from '../../../../config'
 import schema from '../../../../schema'
 import validate from '../../../../validators'
-import SubsectionElement from '../../SubsectionElement'
 import { Name, Field } from '../../../Form'
 
-export default class ApplicantName extends SubsectionElement {
+import { reportCompletion } from '../../../../actions/ApplicationActions'
+
+import connectIdentificationSection from '../IdentificationConnector'
+
+const sectionConfig = {
+  section: 'identification',
+  subsection: 'name',
+  store: 'Identification',
+  storeKey: 'ApplicantName',
+}
+
+export class ApplicantName extends React.Component {
   constructor(props) {
     super(props)
+
+    const { section, subsection, store, storeKey } = sectionConfig
+
+    this.section = section
+    this.subsection = subsection
+    this.store = store
+    this.storeKey = storeKey
+
+    this.handleError = this.handleError.bind(this)
+    this.handleCompletion = this.handleCompletion.bind(this)
     this.update = this.update.bind(this)
     this.updateName = this.updateName.bind(this)
   }
 
+  handleCompletion() {
+    const data = {
+      ...this.props,
+      ...this.state
+    }
+
+    this.props.dispatch(
+      reportCompletion(
+        this.section,
+        this.subsection,
+        this.props.validator(data)
+      )
+    )
+  }
+
+  handleError(value, arr) {
+    this.handleCompletion()
+    arr = arr.map(err => {
+      return {
+        ...err,
+        section: this.section,
+        subsection: this.subsection
+      }
+    })
+    
+    return this.props.onError(value, arr)
+  }
+
```

Comment:
Moved these functions in here in order to decouple from `SubsectionElement`, but ultimately could be extracted and passed into each section component as part of an HOC